src/arcsf/data/data_module.py

```diff
@@ -62,12 +81,25 @@ class EvalQADataset(Dataset):
 
     def __init__(
         self,
-        data,
-        tokenizer,
-        qa_formatter,
-        loss_type,
+        data: Dataset,
+        tokenizer: AutoTokenizer,
+        qa_formatter: callable,
```

Comment:
Instead of `callable` (which is a built-in python function) the typehint should use `Callable` from `collections.abc`: https://docs.python.org/3/library/typing.html#annotating-callable-objects

(i.e. do `from collections.abc import Callable` and then replace all `callable` with `Callable`. You can also type hint the args and return type of the function if you want).

Reply:
Thank you, I wasn't sure how to typehint a function, I've amended this now.